### PR TITLE
PUB-1663 Clear Subscription

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1663

### Change description ###

Create endpoint which will delete all the subscription for the given location. Upon deletion, email will be sent to all the subscribers and system admins about this action.

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/AccountManagementService.java [hunk 1]
```diff
@@ -48,4 +62,57 @@ public Boolean isUserAuthorised(String userId, ListType listType, Sensitivity se
...
+    }
+
+    public List<String> getAllAccounts(String provenances, String role)
+        throws JsonProcessingException {
+        try {
+            String result = webClient.get().uri(String.format(
+                    "%s/account/all?provenances=%s&roles=%s", url, provenances, role))
+                .attributes(clientRegistrationId(ACCOUNT_MANAGEMENT_API))
+                .retrieve().bodyToMono(String.class).block();
+            return findAllSystemAdmins(result);
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/AccountManagementService.java:98]: We should probably rename this method with a generic name. The parent method is generic as you pass in a role (not specific to system admin)

Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/AccountManagementService.java]: agree. rename the method.

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/AccountManagementService.java [hunk 2]
```diff
@@ -48,4 +62,57 @@ public Boolean isUserAuthorised(String userId, ListType listType, Sensitivity se
...
+        } catch (WebClientException ex) {
+            log.error(String.format("Request to account management failed with error message: %s", ex.getMessage()));
+            return List.of("Failed to find all the accounts");
+        }
+    }
+
+    private List<String> findAllSystemAdmins(String result) throws JsonProcessingException {
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/AccountManagementService.java]: I think we should use a model for this here to capture the account return (Happy this being copied into this repo for now with the aim to clean it up later in the overall model refactoring ticket)

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

**Review by KianKwa — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/AccountManagementService.java (hunk 2 above)
Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/AccountManagementService.java]: fixed, now it is using PiUser model.

**Review by ChrisS1512 — APPROVED** [review]

**Review by KianKwa — APPROVED** [review]

**Pull request merged**